### PR TITLE
Fix idempotent issue in grafana_datasource provider

### PR DESCRIPTION
**Pull request by tobias-urdin**
This fixes the idempotent issue mentioned in the issue.

Fixes https://github.com/voxpupuli/puppet-grafana/issues/289

### CODE REVIEW

**Comment by joernott** [comment]
I actually noded, that there are two fields (depending on the datasource)
```
Notice: /Stage[main]/Profiles_monitoring::Grafana::Icinga_config/Grafana_datasource[icinga2_influxdb]/password: changed [redacted] to [redacted]
Notice: /Stage[main]/Profiles_monitoring::Grafana::Icinga_config/Grafana_datasource[icinga2_influxdb]/basic_auth_password: changed [redacted] to [redacted]
```
so, the patch should be applied to the "password" field as well

**Comment by alexjfisher** [comment]
@joernott Good spot.  I've did some manual testing on @tobias-urdin's change and it looked good.

I'll see if I can add some acceptance tests for both types of datasource.

**Comment by alexjfisher** [comment]
This doesn't seem to work if `basic_auth_password` is something other than an empty string (the default).  I _think_ I've seen the issue.  Will test a fix shortly.

File: lib/puppet/provider/grafana_datasource/grafana.rb [hunk 1]
```diff
@@ -63,6 +63,12 @@ def datasources
 
         datasource = JSON.parse(response.body)
 
+        basic_auth_password = if datasource.key?('secureJsonData') && datasource['secureJsonData'].key?('basicAuthPassword')
```
Comment (alexjfisher) [inline lib/puppet/provider/grafana_datasource/grafana.rb:66]: `secureJsonData` is never part of the response.  There's `secureJsonFields` but this just returns booleans.  See https://github.com/grafana/grafana/issues/20274#issuecomment-1064626061 :(

Comment (tobias-urdin) [reply lib/puppet/provider/grafana_datasource/grafana.rb:66]: :(

Comment (tobias-urdin) [reply lib/puppet/provider/grafana_datasource/grafana.rb:66]: We don't use basic auth so I didn't verify that part, but that's bummer. We are using this PR internally until something can be figured out here.

**Pull request closed without merging**